modules/bibupload/lib/bibupload.py

```diff
@@ -1995,11 +1996,8 @@ def _process_document_moreinfos(more_infos, docname, version, docformat, mode):
                             guessed_format = guess_via_magic(downloaded_url)
                             if guessed_format != docformat:
                                 raise RuntimeError("Given URL %s was supposed to refer to format %s but was found to be of format %s. Is this document behind an authentication page?" % (url, docformat, guessed_format))
-                            from PyPDF2 import PdfFileReader
-                            from PyPDF2.utils import PdfReadError
-                            try:
-                                PdfFileReader(open(downloaded_url))
-                            except PdfReadError as err:
+                            exit_code, dummy_out, err = run_shell_command("pdfinfo %s", downloaded_url)
```

Comment:
args has to be a tuple, so 
```
run_shell_command("pdfinfo %s", (downloaded_url,))
```
otherwise you get
```
In [3]: downloaded_url = '/tmp/foo.pdf'

In [4]: run_shell_command("pdfinfo %s", downloaded_url)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-8-db222bea94cc> in <module>()
----> 1 run_shell_command("pdfinfo %s", downloaded_url)

/usr/lib64/python2.6/site-packages/invenio/shellutils.pyc in run_shell_command(cmd, args, filename_out, filename_err)
    116     if args:
    117         try:
--> 118             cmd = cmd % tuple([escape_shell_arg(x) for x in args])
    119         except TypeError:
    120             # there were problems with %s and args substitution, so raise an error:

TypeError: not all arguments converted during string formatting
```
because
```
In [5]: tuple(downloaded_url)
('/', 't', 'm', 'p', '/', 'f', 'o', 'o', '.', 'p', 'd', 'f')
```

Reply:
I can do it. I know I can do it!